src/Sql/Sql/ManagedInstance/Services/AzureSqlManagedInstanceAdapter.cs

```diff
@@ -228,5 +229,27 @@ public static string GetInstanceSkuPrefix(string tier)
 
             return SqlSkuUtils.GetVcoreSkuPrefix(tier) ?? "Unknown";
         }
+
+        /// <summary>
+        /// Get hardware family from SKU name.
+        /// </summary>
+        /// <returns>The model to send to the update</returns>
+        public static string GetHardwareGenerationFromSkuName(string skuName)
```

Comment:
consider moving to SqlSkuUtils.cs

Reply:
Done

Reply:
Note that, in order to run unit tests, SqlSkuUtils had to be changed to public